contracts/cw20-ics20/src/contract.rs

```diff
@@ -0,0 +1,343 @@
+use cosmwasm_std::{
+    attr, entry_point, from_binary, to_binary, Binary, Deps, DepsMut, Env, HumanAddr, IbcMsg,
+    IbcQuery, MessageInfo, Order, PortIdResponse, Response, StdResult,
+};
+
+use cw2::set_contract_version;
+use cw20::{Cw20CoinHuman, Cw20ReceiveMsg};
+
+use crate::amount::Amount;
+use crate::error::ContractError;
+use crate::ibc::Ics20Packet;
+use crate::msg::{
+    ChannelResponse, ExecuteMsg, InitMsg, ListChannelsResponse, PortResponse, QueryMsg, TransferMsg,
+};
+use crate::state::{Config, CHANNEL_INFO, CHANNEL_STATE, CONFIG};
+use cw0::{nonpayable, one_coin};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:cw20-ics20";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn init(
```

Comment:
Note this will conflict with https://github.com/CosmWasm/cosmwasm-plus/pull/243

Reply:
I need that merged in first and a rebase. The contract_checker failed in CI as I didn't have instantiate